pxr/usd/plugin/usdDraco/exportTranslator.cpp

```diff
@@ -0,0 +1,291 @@
+    if (_faceVertexCounts.empty()) {
+        TF_RUNTIME_ERROR("Mesh has no face vertex counts.");
+        return false;
+    }
+    if (_faceVertexIndices.empty()) {
+        TF_RUNTIME_ERROR("Mesh has no face vertex indices.");
+        return false;
+    }
+    if (_positions.GetNumValues() == 0) {
+        TF_RUNTIME_ERROR("Mesh has no points.");
+        return false;
+    }
+    if (!_CheckPrimvarData(_texCoords)) {
+        TF_RUNTIME_ERROR("Mesh texture coordinates index is inconsistent.");
+        return false;
+    }
+    if (!_CheckPrimvarData(_normals)) {
+        TF_RUNTIME_ERROR("Mesh normal index is inconsistent.");
+        return false;
+    }
+    return true;
+}
+
+template <class ArrayT>
+bool UsdDracoExportTranslator::_CheckPrimvarData(
+    const UsdDracoExportAttribute<ArrayT> &attribute) const {
+    if (attribute.GetNumValues() == 0)
+        return true;
+    if (attribute.UsesPositionIndex())
+        return attribute.GetNumIndices() == _positions.GetNumValues();
+    return attribute.GetNumIndices() == _faceVertexIndices.size();
+}
+
+void UsdDracoExportTranslator::_ConfigureGenericAttributes(
+        UsdDracoFlag<bool> preservePolygons,
+        UsdDracoFlag<bool> preservePositionOrder,
+        UsdDracoFlag<bool> preserveHoles) {
+
+    // Conditionally clear position order preservation attribute.
+    if (preservePositionOrder.HasValue()) {
+        if (preservePositionOrder.GetValue() == false) {
+            _posOrder.Clear();
+        }
+    } else {
+        if (!_SubdivisionRefersToFaces()) {
+            _posOrder.Clear();
+        }
+    }
+
+    // Conditionally clear hole faces attribute.
+    if (preserveHoles.HasValue()) {
+        if (preserveHoles.GetValue() == false) {
+            _holeFaces.Clear();
+        }
+    } else {
+        if (!_usdMesh.GetHoleIndicesAttr().HasAuthoredValueOpinion()) {
+            _holeFaces.Clear();
+        }
+    }
+
+    // Conditionally clear polygon preservation attribute.
+    if (preservePolygons.HasValue()) {
+        if (preservePolygons.GetValue() == false) {
+            _addedEdges.Clear();
+        }
+    }
+    if (_HasTrianglesOnly())
+        _addedEdges.Clear();
+}
+
+bool UsdDracoExportTranslator::_HasTrianglesOnly() const {
+    for (size_t i = 0; i < _faceVertexCounts.size(); i++)
+        if (_faceVertexCounts[i] > 3)
+            return false;
+    return true;
+}
+
+bool UsdDracoExportTranslator::_SubdivisionRefersToPositions() const {
+    if (_usdMesh.GetCreaseSharpnessesAttr().HasAuthoredValueOpinion())
+        return true;
+    if (_usdMesh.GetCornerSharpnessesAttr().HasAuthoredValueOpinion())
+        return true;
+    return false;
+}
+
+bool UsdDracoExportTranslator::_SubdivisionRefersToFaces() const {
+    if (_usdMesh.GetHoleIndicesAttr().HasAuthoredValueOpinion())
+        return true;
+    return false;
+}
+
+void UsdDracoExportTranslator::_SetNumPositionsToMesh() const {
+    size_t numPositions = 0;
```

Comment:
I would probably create a function _GetNumTriangles() and then initialize the number of points from that (3 * num_triangles).   Or maybe simply rename this _SetNumPointsToMesh()  .. we don't really have this many position values, only points

Reply:
Done. Renamed the method.